### PR TITLE
chore: merge `emqx-OTP-25.3.2` into `emqx-OTP-26.1.2` and prepare to tag `26.1.2-1`

### PR DESCRIPTION
**Pull request by thalesmg**

### CODE REVIEW

File: lib/kernel/src/gen_tcp.erl [hunk 1]
```diff
@@ -218,21 +223,89 @@ connect(#{family := Fam} = SockAddr, Opts, Timeout)
...
+            maybe_ipv6_1(Host, Opts);
+        false when TryIpv6 ->
+            %% string host
+            maybe_ipv6_2(Host, Opts);
+        false ->
+            Opts
+    end.
+
+maybe_ipv6_1(Ip, Opts) when tuple_size(Ip) =:= 4 -> Opts;
+maybe_ipv6_1(Ip, Opts) when tuple_size(Ip) =:= 8 -> [inet6 | Opts].
```
Comment (zmstone) [inline lib/kernel/src/gen_tcp.erl:287]: Please help to fix this `function_clause` error when a bad size tuple is provided.
Something happens in wolff tests.
```suggestion
maybe_ipv6_1(Ip, Opts) when tuple_size(Ip) =:= 8 -> [inet6 | Opts];
maybe_ipv6_1(_, Opts) -> Opts;
```

Comment (zmstone) [reply lib/kernel/src/gen_tcp.erl:287]:
```
                   {function_clause,
                       [{gen_tcp,maybe_ipv6_1,
                            [{127,0,0},[{active,false},binary]],
                            [{file,"gen_tcp.erl"},{line,278}]},
                        {gen_tcp,connect_maybe_ipv6,6,
                            [{file,"gen_tcp.erl"},{line,237}]},
                        {kpro_connection,connect,4,
                            [{file,
                                 "/mnt/code/src/kafka/wolff/_build/default/lib/kafka_protocol/src/kpro_connection.erl"},
                             {line,224}]},
                        {kpro_connection,init,4,
                            [{file,
                                 "/mnt/code/src/kafka/wolff/_build/default/lib/kafka_protocol/src/kpro_connection.erl"},
                             {line,195}]},
                        {proc_lib,init_p_do_apply,3,
                            [{file,"proc_lib.erl"},{line,240}]}]}},
```

Comment (zmstone) [reply lib/kernel/src/gen_tcp.erl:287]: nvm.
the original code crashes with a function_clause anyway:

```
 {function_clause,[{inet,mod,
                                           [infinity,tcp_module,
                                            {undefined,{127,0,0,1}},
                                            #{local => local_tcp,inet => inet_tcp,inet6 => inet6_tcp},
                                            undefined,[]],
                                           [{file,"inet.erl"},{line,1478}]},
                                     {gen_tcp,connect1,4,[{file,"gen_tcp.erl"},{line,237}]},
                                     {gen_tcp,connect,4,[{file,"gen_tcp.erl"},{line,224}]},
                                     {erl_eval,do_apply,7,[{file,"erl_eval.erl"},{line,750}]},
                                     {shell,exprs,7,[{file,"shell.erl"},{line,782}]},
                                     {shell,eval_exprs,7,[{file,"shell.erl"},{line,738}]},
                                     {shell,eval_loop,4,[{file,"shell.erl"},{line,723}]}]}

```

**Review by zmstone — APPROVED** [review]

**Comment by thalesmg** [comment]
The CI failures are apparently unrelated to the changes herein:

image

**Comment by thalesmg** [comment]
tagged `OTP-26.1.2-1`

**Pull request merged**